src/common/generic.rs

```diff
@@ -0,0 +1,96 @@
+use std::{
+    collections::HashMap,
+    sync::{Arc, Mutex},
+};
+
+use crate::google::protobuf::{value::Kind, Struct, Value};
+
+use super::{
+    config::ConfigType,
+    registry::{ComponentRegistry, Dependency},
+    status::Status,
+};
+
+pub static COMPONENT_NAME: &str = "generic";
+
+pub(crate) fn register_models(registry: &mut ComponentRegistry) {
+    if registry
+        .register_generic_component("fake", &FakeGenericComponent::from_config)
+        .is_err()
+    {
+        log::error!("model fake is already registered")
+    }
+}
+
+pub trait GenericComponent: Status {
+    /// do_command custom commands outside of a strict API. Takes a command struct that can be interpreted
+    /// as a map of method name keys to argument values. Actual interpretation of the command argument
+    /// is completely up to the implementing component struct
```

Comment:
nit: can we rephrase the comment `is completely up to the implementing component struct` doesn't feel right